lighthouse-core/lib/median-run.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * @license Copyright 2019 Google Inc. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/** @param {LH.Result} lhr @param {string} auditName */
+const getNumericValue = (lhr, auditName) =>
+  (lhr.audits[auditName] && lhr.audits[auditName].numericValue) || NaN;
+
+/**
+ * @param {LH.Result} lhr
+ * @param {number} medianFcp
+ * @param {number} medianInteractive
+ */
+function getMedianSortValue(lhr, medianFcp, medianInteractive) {
+  const distanceFcp = medianFcp - getNumericValue(lhr, 'first-contentful-paint');
+  const distanceInteractive =
+    medianInteractive - getNumericValue(lhr, 'interactive');
+
+  return distanceFcp * distanceFcp + distanceInteractive * distanceInteractive;
+}
+
+/**
+ * We want the run that's closest to the median of the FCP and the median of the TTI. 
+ * We're using the squared distance for that (https://en.wikipedia.org/wiki/Euclidean_distance).
+ * @param {Array<LH.Result>} runs
+ * @return {LH.Result|undefined}
+ */
+function computeMedianRun(runs) {
+  const validRuns = runs
+    .filter(run => Number.isFinite(getNumericValue(run, 'first-contentful-paint')))
+    .filter(run => Number.isFinite(getNumericValue(run, 'interactive')));
+  if (!validRuns.length) return undefined;
+
+  const sortedByFcp = validRuns
+    .slice()
+    .sort(
+      (a, b) =>
+        getNumericValue(a, 'first-contentful-paint') -
+        getNumericValue(b, 'first-contentful-paint')
+    );
+  const medianFcp = getNumericValue(
+    sortedByFcp[Math.floor(validRuns.length / 2)],
```

Comment:
since this is looking for the point closest to the middle of the data in each dimension, for the even-number-of-runs case it probably makes sense to do `(sortedByFcp[Math.floor(validRuns.length / 2)] + sortedByFcp[Math.floor(validRuns.length / 2) + 1]) / 2`. That does make the code slightly more annoying, though :)

It doesn't matter in the 1D case (have to pick one or the other by the end regardless), but as dimensions are added it will affect the overall distance (the center point will tend to be closer to the lower values)

Reply:
That was sort of intentional for LHCI (see https://github.com/GoogleChrome/lighthouse/pull/10859#discussion_r432058023), so I'm wondering if you're suggesting this because you think it leads to picking a better run or if you'd like to to be technically faithful to usage of the word median.

If it's the latter, I'd rather add commentary or descriptions than change the impl :)

Reply:
> if you'd like to to be technically faithful to usage of the word median.

I guess it's this one :)

but it's also not clear to me why biasing it for an even number of runs would itself result in a better choice, so it seemed like using the centerpoint in both the even and odd cases keeps things conceptually simpler.

Reply:
Fair, updated the impl 👍